### DISCUSSION TITLE
Using Git Homebrew shim

### DISCUSSION THREAD

**Discussion by ldennington** [Homebrew/discussions#3310] [Writing Formulae/Casks] [answered]
macOS users of [Git Credential Manager](https://github.com/GitCredentialManager/git-credential-manager) (GCM) are encouraged to [install via Homebrew](https://github.com/GitCredentialManager/git-credential-manager/blob/main/README.md#macos-homebrew). However, some users have reported seeing the following error when using brew to upgrade their GCM distributions:

```
fatal: Failed to unset Git configuration entry 'credential.https://dev.azure.com.useHttpPath' [1]
git: This shim is internal and must be run via brew.
```

This seems to be due to the fact that GCM's uninstall script [calls Git](https://github.com/GitCredentialManager/git-credential-manager/blob/8a8005fd31f96365267202929a96f087a536209b/src/osx/Installer.Mac/uninstall.sh#L15) and is run as part of the `upgrade` command. I wanted to reach out to see whether there is any existing guidance to correctly run the Git shim in this type of workflow.

**Comment by ldennington** [comment]
Ping. Wanted to check in and see if anyone in the community has thoughts on this.

**Comment by SMillerDev** [comment, marked as answer]
Does it work if you specify in the script that it needs to run with `/usr/bin/git`?

**Reply by ldennington** [reply]
Thanks! This worked.
